libsnark/gadgetlib1/gadgets/hashes/anemoi/tests/test_anemoi_gadget.cpp

```diff
@@ -339,31 +350,115 @@ void test_anemoi_permutation_mds()
     libff::print_time("anemoi_permutation_mds tests successful");
 }
 
-template<typename ppT> void test_for_curve()
+template<typename ppT> void test_intermediate_gadgets_bls12_381()
 {
-    // Execute all tests for the given curve.
-
-    ppT::init_public_params();
     // Use debug parameters with small values for the small gadgets
     using parameters_debug = parameters_debug_bls12_381;
     test_flystel_Q_gamma_prime_field_gadget<ppT, parameters_debug>();
     test_flystel_Q_gamma_binary_field_gadget<ppT, parameters_debug>();
     test_flystel_E_power_five_gadget<ppT>();
     test_flystel_E_root_five_gadget<ppT, parameters_debug>();
     test_flystel_prime_field_gadget<ppT, parameters_debug>();
+    test_anemoi_permutation_mds_bls12_381();
+}
+
+template<typename ppT>
+void test_for_curve(
+    expected_round_values_fn_t<ppT> expected_round_values_fn = 0)
+{
     // Use the original parameters for the full permutation
     using parameters = anemoi_parameters<ppT>;
-    test_anemoi_permutation_round_prime_field_gadget<ppT, 1, parameters>();
-    test_anemoi_permutation_round_prime_field_gadget<ppT, 2, parameters>();
-    test_anemoi_permutation_round_prime_field_gadget<ppT, 3, parameters>();
-    test_anemoi_permutation_round_prime_field_gadget<ppT, 4, parameters>();
-    test_anemoi_permutation_mds<ppT, parameters>();
+    test_anemoi_permutation_round_prime_field_gadget<ppT, 1, parameters>(
+        expected_round_values_fn);
+    test_anemoi_permutation_round_prime_field_gadget<ppT, 2, parameters>(
+        expected_round_values_fn);
+    test_anemoi_permutation_round_prime_field_gadget<ppT, 3, parameters>(
+        expected_round_values_fn);
+    test_anemoi_permutation_round_prime_field_gadget<ppT, 4, parameters>(
+        expected_round_values_fn);
+}
+
+template<typename ppT> void test_curve_parameters()
```

Comment:
Does this actually test anything?  If we instantiate the gadgets and create a circuite, that's more of a test, so this may be redundant now.

Reply:
A more appropriate name would be `print_curve_parameters` . Anyway, I agree it's redundant now that we have `test_for_curve`, so - removed.